File: robocode2-server/src/main/java/net/robocode2/model/GameState.java
package net.robocode2.model;

import java.util.List;

import lombok.Builder;
import lombok.Singular;
import lombok.Value;

/**
 * Game state.
 * 
 * @author Flemming N. Larsen
 */
@Value
@Builder(toBuilder=true)
public class GameState {

	/** Arena */
	Arena arena;

	/** List of rounds */
	@Singular List<Round> rounds;

	/** Flag specifying if game has ended */
	boolean gameEnded;
	
	/**
	 * Returns the last round.
	 */
	public Round getLastRound() {
		return (rounds == null || rounds.isEmpty()) ? null : rounds.get(rounds.size() - 1);
	}
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/model/GameState.kt
package dev.robocode.tankroyale.server.model

/** Game state */
data class GameState(
    /** Arena */
    val arenaSize: Arena,

    /** List of rounds */
    val rounds: MutableList<IRound> = mutableListOf(),

    /** Flag specifying if game has ended yet */
    var isGameEnded: Boolean = false,
) {
    /** Last round */
    val lastRound: IRound? get() = if (rounds.isNotEmpty()) rounds[rounds.size - 1] else null
}